cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,200 @@
+package nmstate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	sriovtestclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnetwork "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+	nmstateshared "github.com/nmstate/kubernetes-nmstate/api/shared"
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/api/v1"
+	client "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/networks"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var sriovclient *sriovtestclient.ClientSet
+
+const testNamespace string = "nmstate-sriov-testing"
+
+const nmstateDesiredStateTemplateStaticIP string = `interfaces:
+- name: %s
+  description: test for SR-IOV network operator integration
+  type: ethernet
+  state: up
+  ipv4:
+    dhcp: false
+    address:
+    - ip: 192.0.2.2
+      prefix-length: 24
+    enabled: true`
+
+func init() {
+	sriovclient = sriovtestclient.New("")
+}
+
+var _ = Describe("[nmstate] SR-IOV Network Operator Integration", Ordered, func() {
+
+	var sriovInfos *cluster.EnabledNodes
+
+	BeforeAll(func() {
+		if !networks.IsSriovOperatorInstalled() {
+			Skip("SR-IOV operator not installed on the cluster.")
+		}
+
+		err := namespaces.Create(testNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("CleanSriov...")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovInfos, err = cluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(sriovInfos.Nodes)).ToNot(BeZero(), "no node with SR-IOV NICs found")
+
+		nmstate := nmstatev1.NMState{ObjectMeta: metav1.ObjectMeta{Name: "nmstate"}}
+		err = client.Client.Create(context.Background(), &nmstate)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(client.Client.Delete, context.Background(), &nmstate)
+	})
+
+	Context("when a SriovNetworkNodePolicy and NodeNetworkConfigurationPolicySpec target the same NIC", func() {
+
+		BeforeAll(func() {
+			namespaces.CleanPods(testNamespace, client.Client)
+		})
+
+		It("VFs should spawn correctly", func() {
+			node, devices, err := findNonPrimarySriovDevicesAndNode(sriovInfos)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices)).ToNot(BeZero())
+
+			By("Using node " + node)
+
+			testDevice := devices[0]
+			By("Using device " + testDevice.Name)
+
+			By("Configuring IPAM on Physical Function via NMState")
```

Comment:
Is that different than  "Configuring and IPv4 address"? just IPAM (Address Management) makes me think whereabouts/or other IPAM is being deployed

Reply:
IPv4 is definitely more clear! changing